sdk/servicebus/Azure.Messaging.ServiceBus/src/Processor/ReceiverManager.cs

```diff
@@ -0,0 +1,358 @@
+        /// <returns></returns>
+        public async Task ProcessOneMessage(
+            ServiceBusReceivedMessage message,
+            CancellationToken processorCancellationToken)
+        {
+            ServiceBusErrorSource errorSource = ServiceBusErrorSource.Receive;
+            CancellationTokenSource renewLockCancellationTokenSource = null;
+            Task renewLock = null;
+
+            try
+            {
+                if (!Receiver.IsSessionReceiver &&
+                    Receiver.ReceiveMode == ReceiveMode.PeekLock &&
+                    AutoRenewLock)
+                {
+                    renewLockCancellationTokenSource = CancellationTokenSource.CreateLinkedTokenSource(processorCancellationToken);
+                    renewLock = RenewMessageLock(
+                        message,
+                        renewLockCancellationTokenSource);
+                }
+
+                errorSource = ServiceBusErrorSource.UserCallback;
+
+                await OnMessageHandler(message, processorCancellationToken).ConfigureAwait(false);
+
+                if (Receiver.ReceiveMode == ReceiveMode.PeekLock &&
+                    _processorOptions.AutoComplete &&
+                    !message.IsSettled)
+                {
+                    errorSource = ServiceBusErrorSource.Complete;
+                    // don't pass the processor cancellation token
+                    // as we want in flight autocompletion to be able
+                    // to finish
+                    await Receiver.CompleteAsync(
+                        message.LockToken,
+                        CancellationToken.None)
+                        .ConfigureAwait(false);
+                }
+
+                await CancelTask(renewLockCancellationTokenSource, renewLock).ConfigureAwait(false);
+            }
+            catch (Exception ex) when (!(ex is TaskCanceledException))
+            {
+                ThrowIfSessionLockLost(ex, errorSource);
+                await RaiseExceptionReceived(
+                    new ProcessErrorEventArgs(
+                        ex,
+                        errorSource,
+                        _fullyQualifiedNamespace,
+                        _entityPath))
+                    .ConfigureAwait(false);
+
+                // if the user settled the message, or if the message or session lock was lost,
+                // do not attempt to abandon the message
+                ServiceBusException.FailureReason? failureReason = (ex as ServiceBusException)?.Reason;
+                if (!message.IsSettled &&
+                    _receiverOptions.ReceiveMode == ReceiveMode.PeekLock &&
+                    failureReason != ServiceBusException.FailureReason.SessionLockLost &&
+                    failureReason != ServiceBusException.FailureReason.MessageLockLost)
+                {
+                    try
+                    {
+                        await Receiver.AbandonAsync(message.LockToken).ConfigureAwait(false);
```

Comment:
pass ct?

Reply:
Actually, I think it is correct to not pass it. We have a contract in the processor that if the user handler throws we will abandon the message. Even if the user stops processing, we should still fulfill our obligation. This is similar to the autocompletion handling above.

Reply:
Added a comment to that effect.